### ISSUE TITLE
poseidon 3262 issues - only 1 temp sensor is working from 6

### ISSUE THREAD

**Issue by geertcools** [Noles/check_mk_plugins#1]
I have a poseidon 3262

I needed to change some script code, to get the plugin to work.
only problem I now have , is that only the first temperature check is giving a value back
# 
# TEMP
# 

poseidon_sensors_temp_default_levels = (35, 40)

def inventory_poseidon_sensors_temp(info):
    inventory = []
    for sensor_name, sensor_value, sensor_type, sensor_unit in info:
                **if sensor_type == "0":**
                        inventory.append((sensor_name, poseidon_sensors_temp_default_levels ))
    return inventory

def check_poseidon_sensors_temp(item, params, info):
    **warn, crit = params**
    for sensor_name, sensor_value, sensor_type, sensor_unit in info:
        **if sensor_name == item:
            temp = savefloat(sensor_value)/10
            perfdata = [ ( "temp", temp, warn, crit ) ]
        if temp > crit:
          return (2, "CRITICAL - " + str(temp) +  " C" , perfdata)
        if temp > warn:
          return (1, "WARNING - "  + str(temp) +  " C", perfdata)
        else:
          return (0, "OK - " + str(temp) +  " C", perfdata)**
    return (3, "Sensor not found in SNMP data")

check_info["poseidon_sensors.temp"] = {
    "check_function"        : check_poseidon_sensors_temp,
    "inventory_function"    : inventory_poseidon_sensors_temp,
    "service_description"   : "Temperature %s",
    "group"                 : "hw_temperature",
    "has_perfdata"          : True,
    "snmp_info"             : poseidon_sensors_info,
    "snmp_scan_function"    : poseidon_sensors_scan,
    "includes"              : [ "temperature.include" ],
}

**Comment by Noles** [comment]
Do you have other sensor types than temp and hum?
Could you please attach the output of `snmpwalk xxxx 1.3.6.1.4.1.21796.3.3.3.1`

**Comment by geertcools** [comment]
Hi Noles

As you see I only have temp sensors.

SNMPv2-SMI::enterprises.21796.3.3.3.1.2.1 = STRING: "zonboiler_in"
SNMPv2-SMI::enterprises.21796.3.3.3.1.2.2 = STRING: "zonboiler_uit"
SNMPv2-SMI::enterprises.21796.3.3.3.1.2.3 = STRING: "warm_water"
SNMPv2-SMI::enterprises.21796.3.3.3.1.2.4 = STRING: "vloer_warm"
SNMPv2-SMI::enterprises.21796.3.3.3.1.2.5 = STRING: "lucht_buiten_in"
SNMPv2-SMI::enterprises.21796.3.3.3.1.2.6 = STRING: "lucht_huis_in"
SNMPv2-SMI::enterprises.21796.3.3.3.1.4.1 = INTEGER: 1
SNMPv2-SMI::enterprises.21796.3.3.3.1.4.2 = INTEGER: 1
SNMPv2-SMI::enterprises.21796.3.3.3.1.4.3 = INTEGER: 1
SNMPv2-SMI::enterprises.21796.3.3.3.1.4.4 = INTEGER: 1
SNMPv2-SMI::enterprises.21796.3.3.3.1.4.5 = INTEGER: 1
SNMPv2-SMI::enterprises.21796.3.3.3.1.4.6 = INTEGER: 1
SNMPv2-SMI::enterprises.21796.3.3.3.1.5.1 = STRING: "21.2 C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.5.2 = STRING: "21.0 C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.5.3 = STRING: "25.5 C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.5.4 = STRING: "23.0 C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.5.5 = STRING: "14.8 C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.5.6 = STRING: "18.8 C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.6.1 = INTEGER: 212
SNMPv2-SMI::enterprises.21796.3.3.3.1.6.2 = INTEGER: 210
SNMPv2-SMI::enterprises.21796.3.3.3.1.6.3 = INTEGER: 255
SNMPv2-SMI::enterprises.21796.3.3.3.1.6.4 = INTEGER: 230
SNMPv2-SMI::enterprises.21796.3.3.3.1.6.5 = INTEGER: 148
SNMPv2-SMI::enterprises.21796.3.3.3.1.6.6 = INTEGER: 188
SNMPv2-SMI::enterprises.21796.3.3.3.1.7.1 = INTEGER: 212
SNMPv2-SMI::enterprises.21796.3.3.3.1.7.2 = INTEGER: 210
SNMPv2-SMI::enterprises.21796.3.3.3.1.7.3 = INTEGER: 255
SNMPv2-SMI::enterprises.21796.3.3.3.1.7.4 = INTEGER: 230
SNMPv2-SMI::enterprises.21796.3.3.3.1.7.5 = INTEGER: 148
SNMPv2-SMI::enterprises.21796.3.3.3.1.7.6 = INTEGER: 188
SNMPv2-SMI::enterprises.21796.3.3.3.1.8.1 = INTEGER: 37860
SNMPv2-SMI::enterprises.21796.3.3.3.1.8.2 = INTEGER: 27198
SNMPv2-SMI::enterprises.21796.3.3.3.1.8.3 = INTEGER: 33985
SNMPv2-SMI::enterprises.21796.3.3.3.1.8.4 = INTEGER: 26777
SNMPv2-SMI::enterprises.21796.3.3.3.1.8.5 = INTEGER: 35579
SNMPv2-SMI::enterprises.21796.3.3.3.1.8.6 = INTEGER: 33047
SNMPv2-SMI::enterprises.21796.3.3.3.1.9.1 = INTEGER: 0
SNMPv2-SMI::enterprises.21796.3.3.3.1.9.2 = INTEGER: 0
SNMPv2-SMI::enterprises.21796.3.3.3.1.9.3 = INTEGER: 0
SNMPv2-SMI::enterprises.21796.3.3.3.1.9.4 = INTEGER: 0
SNMPv2-SMI::enterprises.21796.3.3.3.1.9.5 = INTEGER: 0
SNMPv2-SMI::enterprises.21796.3.3.3.1.9.6 = INTEGER: 0
SNMPv2-SMI::enterprises.21796.3.3.3.1.10.1 = STRING: "C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.10.2 = STRING: "C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.10.3 = STRING: "C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.10.4 = STRING: "C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.10.5 = STRING: "C"
SNMPv2-SMI::enterprises.21796.3.3.3.1.10.6 = STRING: "C"

**Comment by geertcools** [comment]
It seems that only the first value is read correct, and the others are not

![image](https://cloud.githubusercontent.com/assets/6011989/15454326/e3fb9fac-2035-11e6-990e-1096fb5f569e.png)

**Comment by geertcools** [comment]
Hi Noles,

found my issue :-)
I was not aware, that for Perl the spaces and place for your command are so important :-|

![image](https://cloud.githubusercontent.com/assets/6011989/15455054/1be5a026-2049-11e6-8516-a7ce63ea2dea.png)

This is what I ended with :)
[temp_check.txt](https://github.com/Noles/check_mk_plugins/files/276414/temp_check.txt)

**Comment by Noles** [comment]
I have updated the sensor type to be more specific. Regarding  return... is it really needed to replace the 
`check_temperature(int(sensor_value), params)`
call? Because this should be used to follow the check_mk guidelines.

**Comment by geertcools** [comment]
Hi Alexander,

I'm definitely not a programmer, so ? :-)

I used this page and some examples from other plugins to find what is
working for me know.
https://mathias-kettner.de/checkmk_devel_snmpbased.html

In your return check_temperature(int(sensor_value), params), there is no
perfdata or i'm I missing something?

Grz,
Geert

On Thu, May 26, 2016 at 2:32 AM, Alexander Korinek <notifications@github.com

**Comment by Noles** [comment]
check_temperature returns perfdata as define in `temperature.include.`

`temperature.include` is used for many other check_mk shipped temperatur checks already.

```
Humidity RaumHum_Rack C OK - 34 %RH  (warn/crit below 14.0/13.0 or above 60.0/65.0 %) (humidity=34;13;14;60;65)
Temperature Rack C1  OK - 26 °C                                              (temp=26;35;40;;)
Temperature RaumTemp_Rack C OK - 26 °C                                       (temp=26;35;40;;)
```

Do you get errors with the last update?

**Comment by geertcools** [comment]
Hi Alexander,

I'm still getting the same issue when I use the last update

![image](https://cloud.githubusercontent.com/assets/6011989/15800709/1a12ad6a-2a81-11e6-8a0c-7b58af83fff3.png)

**Comment by Noles** [comment]
Should be fixed now with version 1.2
